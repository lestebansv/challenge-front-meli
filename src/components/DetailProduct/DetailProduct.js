import CurrencyFormat from 'react-currency-format';
import styles from './DetailProduct.module.scss'

/**
 * Component builds the product description
 * 
 * @param data (JSON) - Product Information
 * @returns Render view product detail
 */
const DetailProduct = (data) => {

    const imgSize = data.data.img[0].size.split('x');
    const condition = data.data.condition.charAt(0).toUpperCase() + data.data.condition.slice(1);

    return (
        <div className={styles.card}>

            <div className={styles.item_image}>
                <img src={data.data.img[0].url} width={imgSize[0]} height={imgSize[1]} alt={data.data.tags}></img>
            </div>

            <div className={styles.info}>
                <div className={styles.condition}>{condition} - {data.data.sold_quantity} vendidos</div>
                <div className={styles.title}>{data.data.title}</div>
                <div className={styles.price}><CurrencyFormat value={data.data.price.amount} displayType={'text'} thousandSeparator={true} prefix={'$ '} /></div>
                <button className={styles.button}>Comprar</button>
            </div>

            <div className={styles.description}>
                <p className={styles.title_description}>Descripci√≥n del producto</p>
                <div className={styles.text_description}>{data.data.description}</div>
            </div>
        </div>
    )
}

export default DetailProduct;