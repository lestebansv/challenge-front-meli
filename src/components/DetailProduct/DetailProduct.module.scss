@import '../../assets/css/styles.scss';

.button {
    background-color: $background-blue;
    border: 0px;
    border-radius: 5px;
    color: $background-white;
    cursor: pointer;
    height: 33px;
    width: 100%;
}

.card {
    background-color: $background-white;
    border-radius: 4px 4px 0 0;
    display: grid;
    grid-auto-rows: minmax(100px, auto);
    grid-template-columns: 70% 1fr;
    margin: 0 15%;
    margin-bottom: 30px;
}

.condition {
    font-size: 14px;
    margin-bottom: 16px;
    margin-top: 36px;
}

.description {
    grid-column: 1 / 3;
    grid-row: 2;
    margin-left: 32px;
    margin-top: 40px;
    max-width: 680px;
}

.info {
    grid-column: 2;
    grid-row: 1;
    margin-right: 32px;
}

.item_image {
    grid-column: 1;
    grid-row: 1;
    margin-left: 32px;
    margin-top: 32px;
    max-width: 680px;
    text-align: center;
}

.price {
    font-size: 46px;
    font-weight: 600;
    margin-bottom: 32px;
}

.text_description {
    font-size: 16px;
    margin-bottom: 32px;
    margin-right: 32px;
    max-width: 550px;
    text-align: justify;
}

.title {
    font-size: 24px;
    font-weight: 600;
    margin-bottom: 32px;
}

.title_description {
    font-size: 28px;
}