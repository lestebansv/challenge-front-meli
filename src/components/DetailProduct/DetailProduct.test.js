import { render } from '@testing-library/react';
import DetailProduct from './DetailProduct';

describe('DetailProduct', () => {

    it('validate renders search item kindle', () => {

        const productDetail = {
            id: 'MLA1135260991',
            title: 'E-reader  Kindle Paperwhite 10 Gen 8gb Azul Con Pantalla De 6  300ppp',
            price: {
                currency: 'ARS',
                amount: 35650
            },
            picture: 'http://http2.mlstatic.com/D_674706-MLA40690247901_022020-I.jpg',
            condition: 'new',
            free_shipping: true,
            sold_quantity: 100,
            tags: [
                'good_quality_picture',
                'good_quality_thumbnail',
                'extended_warranty_eligible',
                'immediate_payment',
                'cart_eligible'
            ],
            description: 'Con este E-reader Kindle, podrás almacenar los libros que más te gustan y llevarlos a todos lados.\n A diferencia de otros dispositivos digitales, los libros electrónicos tienen una tinta electrónica que nos permite leer con una calidad similar a la del papel, sin demasiada fatiga visual. Además, este tipo de pantalla consume menos energía, por lo tanto, la batería puede durar días. ¡Volá con tu imaginación donde sea que estés!\n\nMúltiples formatos\nEste dispositivo soporta una gran variedad de formatos de archivos, lo que te permitirá acceder a una mayor cantidad de textos.\n\nCuida tu vista\nEste E-reader cuenta luz de lectura que se distribuye por la pantalla para no impactar directamente en tu vista. Esto te permitirá leer durante largos períodos de tiempo sin sentir cansancio en tus ojo.\n\nAccesibilidad y comodidad\n¿Te gusta hacer actividades mientras escuchás un libro? Ahora es posible, ya que este dispositivo es compatible con audiolibros. Esto no solo brinda comodidad para quienes prefieren escuchar los textos, sino que garantiza una mayor accesibilidad para personas ciegas o con disminución visual.',
            img: [
                {
                    id: '674706-MLA40690247901_022020',
                    url: 'http://http2.mlstatic.com/D_674706-MLA40690247901_022020-O.jpg',
                    secure_url: 'https://http2.mlstatic.com/D_674706-MLA40690247901_022020-O.jpg',
                    size: '348x500',
                    max_size: '1093x1568',
                    quality: ''
                },
                {
                    id: '999603-MLA40690247588_022020',
                    url: 'http://http2.mlstatic.com/D_999603-MLA40690247588_022020-O.jpg',
                    secure_url: 'https://http2.mlstatic.com/D_999603-MLA40690247588_022020-O.jpg',
                    size: '494x500',
                    max_size: '1114x1126',
                    quality: ''
                }
            ]
        };

        const { container } = render(<DetailProduct data={productDetail} />);

        expect(container).toMatchSnapshot();
    });

})