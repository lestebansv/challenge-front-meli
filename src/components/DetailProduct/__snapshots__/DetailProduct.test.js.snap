// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailProduct validate renders search item kindle 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="item_image"
    >
      <img
        alt="good_quality_picture,good_quality_thumbnail,extended_warranty_eligible,immediate_payment,cart_eligible"
        height="500"
        src="http://http2.mlstatic.com/D_674706-MLA40690247901_022020-O.jpg"
        width="348"
      />
    </div>
    <div
      class="info"
    >
      <div
        class="condition"
      >
        New
         - 
        100
         vendidos
      </div>
      <div
        class="title"
      >
        E-reader  Kindle Paperwhite 10 Gen 8gb Azul Con Pantalla De 6  300ppp
      </div>
      <div
        class="price"
      >
        <span>
          $ 35,650
        </span>
      </div>
      <button
        class="button"
      >
        Comprar
      </button>
    </div>
    <div
      class="description"
    >
      <p
        class="title_description"
      >
        Descripción del producto
      </p>
      <div
        class="text_description"
      >
        Con este E-reader Kindle, podrás almacenar los libros que más te gustan y llevarlos a todos lados.
 A diferencia de otros dispositivos digitales, los libros electrónicos tienen una tinta electrónica que nos permite leer con una calidad similar a la del papel, sin demasiada fatiga visual. Además, este tipo de pantalla consume menos energía, por lo tanto, la batería puede durar días. ¡Volá con tu imaginación donde sea que estés!

Múltiples formatos
Este dispositivo soporta una gran variedad de formatos de archivos, lo que te permitirá acceder a una mayor cantidad de textos.

Cuida tu vista
Este E-reader cuenta luz de lectura que se distribuye por la pantalla para no impactar directamente en tu vista. Esto te permitirá leer durante largos períodos de tiempo sin sentir cansancio en tus ojo.

Accesibilidad y comodidad
¿Te gusta hacer actividades mientras escuchás un libro? Ahora es posible, ya que este dispositivo es compatible con audiolibros. Esto no solo brinda comodidad para quienes prefieren escuchar los textos, sino que garantiza una mayor accesibilidad para personas ciegas o con disminución visual.
      </div>
    </div>
  </div>
</div>
`;
